packages/react/src/components/FileUploader/index.js

```diff
@@ -25,45 +25,54 @@ const texts = {
     id: "flamingo-file-uploader-error-try-again",
     defaultText: "Click here to try again",
   },
+  upload_more_files: {
+    id: "flamingo-file-uploader-upload-more-files",
+    defaultText: "Add more files",
+  },
 };
 
 const FileUploader = ({
   children,
   className,
-  file,
+  files: filesProp,
   name,
   onChange,
   translate,
   isLoading,
   hasError,
+  multiple,
   overrides,
   ...props
 }) => {
-  const [innerFile, setInnerFile] = React.useState(undefined);
-  const hasFile = !isLoading && (file || innerFile);
+  const [files, setFiles] = React.useState(filesProp);
+  const hasFile = !isLoading && !hasError && files.length > 0;
 
-  const handleFileChange = e => {
-    const [eventFile] = e.target.files;
+  const handleFilesChange = e => {
+    const inputFiles = [...e.target.files];
+    const nextFiles = multiple ? [...files, ...inputFiles] : inputFiles;
 
-    setInnerFile(eventFile);
-    safeInvoke(onChange(eventFile));
+    setFiles(nextFiles);
+    safeInvoke(onChange(inputFiles));
   };
 
-  const handleResetState = () => {
-    setInnerFile(undefined);
+  const handleDeleteFile = fileToDelete => {
+    const filteredFiles = files.filter(file => file.name !== fileToDelete.name);
```

Comment:
```suggestion
    const remainingFiles = files.filter(file => file.name !== fileToDelete.name);
```
?

No strong opinion though

Reply:
✅